### DISCUSSION TITLE
Disk space error

### DISCUSSION THREAD

**Discussion by nmassey** [renovatebot/renovate#21960] [Request Help] [answered]
### I'm using the Mend Renovate hosted app on github.com.

- [X] I'm using the Mend Renovate hosted app.

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe your problem.

A recent job ID for us is `1063506734`.

Up until last week sometime (perhaps around April 27? Unsure because our renovate logs don't go back that far 😢), Renovate was working great for us!

But, now, every job shows result = `caught-error`. Looking into the logs, it seems that upon running `yarn`, we run out of disk space.

AFAIK, our repository has not grown substantially in the past week. ... 🤔
(Our monorepo directory is around 500 MB total after clone; approx. 2.9 GB total after running `yarn` locally. But perhaps we just happened to cross some threshold?)

Thank you so much for any help you may be able to provide! And thanks again for your work on Renovate! We love it. 🙏❤️

### Relevant debug logs.

<details><summary>Logs</summary>

## from log - lines 178117-178119

```
DEBUG: Passing disk-space error up(branch="renovate/prettier-2.x-lockfile")
ERROR: Disk space error - skipping
DEBUG: Repository result: disk-space, status: onboarded, enabled: true, onboarded: true
```

## partial yarn output (from stdout from `DEBUG: rawExec err(branch="renovate/prettier-2.x-lockfile")` - line 178054)

```
➤ YN0000: └ Completed in 1s 545ms
➤ YN0000: ┌ Fetch step
➤ YN0013: │ 42 packages were already cached, 3264 had to be fetched
➤ YN0000: └ Completed in 3m 7s
➤ YN0000: ┌ Link step
➤ YN0001: │ Error: While cloning /mnt/renovate/gh/.../node_modules/stylehacks/node_modules/postcss -> /mnt/renovate/gh/.../node_modules/postcss-flexbugs-fixes/node_modules/postcss ENOSPC: no space left on device, copyfile '/mnt/renovate/gh/.../node_modules/stylehacks/node_modules/postcss/LICENSE' -> '/mnt/renovate/gh/.../node_modules/postcss-flexbugs-fixes/node_modules/postcss/LICENSE'
➤ YN0000: └ Completed in 2m 1s
➤ YN0000: Failed with errors in 5m 9s
```

</details>

**Comment by rarkins** [comment]
What was the timestamp of the last successful _commit_ in a Renovate PR?

**Reply by nmassey** [reply]
We had:
1. a **proper new Renovate PR** (our PR num 1822, commit `6b700d2`) created on April 18, 1:45pm Pacific
2. a force-push for a Renovate PR (our PR num 1801, commit `ce77901`) at April 27, 12:44:24pm Pacific - `Thu Apr 27 19:44:24 2023 +0000`
3. a force-push for a Renovate PR (our PR num 1588, commit `4bee97f`) at (yesterday) May 3, 2:54:30pm Pacific - `Wed May 3 21:54:30 2023 +0000` - _**note that this Renovate run failed** (same error as above) even though this force-push occurred_

**Reply by nmassey** [reply]
Ooooo Renovate just ran **successfully**! job IDs `1064534853` and `1064535058` ✔️

(Did you change anything? If so, thank you!!)

One tiny difference in the logs is old localDir was `/mnt/renovate/gh/...`, new localDir is `/tmp/renovate/gh/...`

**Reply by rarkins** [reply, marked as answer]
Yes, I switched you from the default SSD /mnt (which has ~4GB) to the ~20GB EBS /tmp. Well spotted!

**Reply by nmassey** [reply]
Amazing. Thanks for doing this. Super helpful!! 🙌

Take care. 🙏
